packages/manager/apps/web/client/app/hosting/multisite/diagnostic/dialog.controller.js

```diff
@@ -1,34 +1,47 @@
-import { DIAGNOSTIC_STATE } from '../hosting-multisite.constants';
+import { RECORD_TYPE_TO_HOSTING_IP } from '../hosting-multisite.constants';
 
 export default class MultisiteDiagnosticCtrl {
   /* @ngInject */
-  constructor($scope, coreURLBuilder) {
-    this.scope = $scope;
+  constructor($scope, coreURLBuilder, HostingDomain, coreConfig) {
+    this.$scope = $scope;
     this.coreURLBuilder = coreURLBuilder;
+    this.HostingDomain = HostingDomain;
+    this.currentNic = coreConfig.getUser().auth.account;
 
-    const { options } = $scope.currentActionData;
-    this.ip = options.ip;
-    this.domain = options.domain;
-    this.dns = options.dns;
-    this.nic = options.nic;
-    this.recordType = options.recordType;
-    this.isDnsExternal = options.isDnsExternal;
-    this.diagnosticState = options.diagnosticState;
-    this.isDnsAttachedToNic = options.isDnsAttachedToNic;
-
-    const { zone } = options;
-    this.linkToARecordModification = this.coreURLBuilder.buildURL(
-      'web',
-      '#/domain/:zone/zone',
-      {
-        zone,
-      },
-    );
-
-    this.DIAGNOSTIC_STATE = DIAGNOSTIC_STATE;
+    const { digStatus, recordType } = $scope.currentActionData;
+    this.digStatus = digStatus;
+    this.domain = digStatus.domain;
+    this.zone = digStatus.zone;
+    this.recordType = recordType;
+    this.ip = digStatus[recordType]?.ip;
+    this.hostingIp = this.$scope.hosting[
+      RECORD_TYPE_TO_HOSTING_IP[this.recordType]
+    ];
+    this.isOVHZone = digStatus.isOVHZone;
+    this.canChangeDns = false;
   }
 
-  resetAction() {
-    this.scope.resetAction();
+  $onInit() {
+    this.loading = true;
+    if (this.isOVHZone) {
+      this.linkToARecordModification = this.coreURLBuilder.buildURL(
+        'web',
+        '#/domain/:zone/zone',
+        {
+          zone: this.zone,
+        },
+      );
+      this.HostingDomain.getZoneServiceInfos(this.zone).then(
+        ({ contactAdmin, contactTech }) => {
+          this.canChangeDns = [contactAdmin, contactTech].includes(
+            this.currentNic,
+          );
+          this.contactAdmin = contactAdmin;
+          this.loading = false;
+        },
+      );
+    } else {
+      this.loading = false;
+    }
```

Comment:
can be simplified and what about catching error ?
```suggestion
  if (this.isOVHZone) {
      this.loading = true;
      this.linkToARecordModification = this.coreURLBuilder.buildURL(
        'web',
        '#/domain/:zone/zone',
        {
          zone: this.zone,
        },
      );
      this.HostingDomain.getZoneServiceInfos(this.zone).then(
        ({ contactAdmin, contactTech }) => {
          this.canChangeDns = [contactAdmin, contactTech].includes(
            this.currentNic,
          );
          this.contactAdmin = contactAdmin;
          this.loading = false;
        },
      );
    } 
```

Reply:
It's fixed, i added the error case for don't purpose redirect to dns modification if there is an error but no need error message.